SocketRocket/SRWebSocket.m

```diff
@@ -651,18 +651,33 @@ - (void)_writeData:(NSData *)data;
 
 - (void)send:(id)data;
 {
+    if (!data) {
+        [self sendData:nil]; // Send Data, but it doesn't matter since we are going to send the same text frame with 0 length.
+    } else if ([data isKindOfClass:[NSString class]]) {
+        [self sendString:data];
+    } else if ([data isKindOfClass:[NSData class]]) {
+        [self sendData:data];
+    } else {
+        NSAssert(NO, @"Unrecognized message. Not able to send anything other than a String or NSData.");
+    }
+}
+
+- (void)sendString:(NSString *)string
+{
+    NSAssert(self.readyState != SR_CONNECTING, @"Invalid State: Cannot call send: until connection is open");
+    dispatch_async(_workQueue, ^{
+        [self _sendFrameWithOpcode:SROpCodeTextFrame data:[string dataUsingEncoding:NSUTF8StringEncoding]];
+    });
+}
+
+- (void)sendData:(NSData *)data
+{
     NSAssert(self.readyState != SR_CONNECTING, @"Invalid State: Cannot call send: until connection is open");
```

Comment:
We should still copy data... If it's a NSMutableData, doing this async means that unintended changes after a call to `sendData:` has been made. Same with `sendString:`.